stubs/seaborn/seaborn/axisgrid.pyi

```diff
@@ -221,22 +378,22 @@ def pairplot(
     size: float | None = None,  # deprecated
 ) -> PairGrid: ...
 def jointplot(
-    data: Incomplete | None = None,
+    data: DataSource | _DataSourceWideForm | None = None,
     *,
-    x: Incomplete | None = None,
-    y: Incomplete | None = None,
-    hue: Incomplete | None = None,
-    kind: str = "scatter",  # ideally Literal["scatter", "kde", "hist", "hex", "reg", "resid"] but it is checked with startswith
+    x: ColumnName | _Vector | None = None,
+    y: ColumnName | _Vector | None = None,
+    hue: ColumnName | _Vector | None = None,
+    kind: Literal["scatter", "kde", "hist", "hex", "reg", "resid"] = "scatter",
```

Comment:
This was changed to `str` earlier thanks to the [feedback](https://github.com/python/typeshed/pull/10721#discussion_r1355999012) on the PR that introduced seaborn stubs. Upon further investigation, the function seems to [only accepts this set of strings](https://github.com/mwaskom/seaborn/blob/908ca95137c0e73bb6ac9ce9a8051577b6453138/seaborn/axisgrid.py#L2223) (also verified in the repl).